### DISCUSSION TITLE
warn  - Invalid next.config.js options detected

### DISCUSSION THREAD

**Discussion by richleach** [vercel/community#1066] [Help] [answered]
I'm trying to get a Next.js build out the door and during my deployment I get the warning: Invalid next.config.js options detected. 

I made the correction to the file, saved and pushed to my github repo (I verified the changed file is in the main branch) but there still seems to be some kind of caching going on, is there something I need to do to ensure there is no caching of files from my github repo during build? I'm using my connected Git repo which is in fact, pointing to the correct repo and version. 

During my builds I am NOT redeploying with existing build cache (that checkbox is unchecked).

Any ideas on why it seems like caching is taking place during my builds?

Thank you,

Rich

**Comment by amyegan** [comment, marked as answer]
Hi @richleach! Is your git repo public, or can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)? I assume you are using the [Vercel for GitHub](https://vercel.com/docs/concepts/git/vercel-for-github) integration, but please let me know if that is not the case.

**Reply by amyegan** [reply]
Another idea that came up in a different discussion is monorepo caching. Are you using something like Turborepo or Nx?

**Reply by richleach** [reply]
Sorry for my delayed response Amy, holidays....
To answer your questions I am not using any monorepo caching (that I'm aware of). This is a straight up create-next-app and yes I am using Vercel for Github integration. Kinda scratching my head on this one, I suppose I could just create a brand new git repo and try building a new Vercel hosted site? I already tried loading files from a different repo, my limited debugging didn't yield any results yet....
Thank you for helping me, I appreciate you.
Rich

**Reply by richleach** [reply]
fyi my github repo 
https://github.com/richleach/cfsnap-cms

**Reply by amyegan** [reply]
Thanks for sharing the repo! I cloned a copy of it to test. I was getting a lot of Typescript errors, so I added `typescript: {ignoreBuildErrors:true}` to `next.config.js`. Now the build results in the following error:
```
10:11:15.765 | info  - Skipping validation of types
10:11:15.765 | info  - Skipping linting
10:11:16.189 | info  - Creating an optimized production build...
10:11:20.943 | Failed to compile.
10:11:20.943 |  
10:11:20.943 | Module not found: Error: Can't resolve 'private-next-pages/blog/' in '/vercel/path0'
10:11:20.944 |  
10:11:20.944 |  
10:11:20.944 | > webpack config.resolve.alias was incorrectly overridden. https://nextjs.org/docs/messages/invalid-resolve-alias
10:11:20.963 | error Command failed with exit code 1.
10:11:20.963 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
10:11:20.987 | Error: Command "yarn run build" exited with 1
```

I noticed that Yarn is insisting on updating some packages in the lockfile, so I wonder if the difference in errors is because of some dependencies that need to be updated. I have not been able to reproduce any warnings about invalid `next.config.js` options.

**Comment by richleach** [comment]
Thank you Amy.
Before I kill any more time I'm going to simply rebuild making sure I only use NPM, I have a feeling I may have used both Yarn and NPM during this project. I'll try again and let you know how it goes, and thank you again for your help.
Rich
